src/collapse/xctrace.rs

```diff
@@ -0,0 +1,455 @@
+use quick_xml::{
+    events::{attributes::Attributes, Event},
+    reader::Reader,
+};
+use std::{
+    borrow::Cow,
+    collections::BTreeMap,
+    io::{self, BufRead},
+    sync::Arc,
+};
+
+use super::{
+    common::{fix_partially_demangled_rust_symbol, Occurrences},
+    Collapse,
+};
+
+const REF: &[u8] = b"ref";
+const ID: &[u8] = b"id";
+const NAME: &[u8] = b"name";
+const TRACE_QUERY_RESULT: &[u8] = b"trace-query-result";
+const NODE: &[u8] = b"node";
+const ROW: &[u8] = b"row";
+const BACKTRACE: &[u8] = b"backtrace";
+const FRAME: &[u8] = b"frame";
```

Comment:
It'd be really useful to have a very brief comment for each of this for what that kind of element _means_ in xctrace.